frontend/pages/sites/$siteId/storage/NewFileOrFolder.jsx

```diff
@@ -0,0 +1,103 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import FileUpload from '@shared/FileUpload';
+import { IconFolder } from '@shared/icons';
+
+const NewFileOrFolder = ({ onUpload, onCreateFolder }) => {
+  const [folderName, setFolderName] = useState('');
+  const [creatingFolder, setCreatingFolder] = useState(false);
+  const [showFolderNameField, setShowFolderNameField] = useState(false);
+  const [showFileDropZone, setShowFileDropZone] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const handleCreateFolder = async () => {
+    const trimmedName = folderName.trim();
+
+    if (trimmedName.length < 1) {
+      return;
+    }
+
+    setCreatingFolder(true);
+    setErrorMessage('');
+
+    try {
+      await onCreateFolder(trimmedName);
+      setFolderName('');
+      setShowFolderNameField(false);
+    } catch (error) {
+      setErrorMessage(error.message || 'Failed to create folder.');
```

Comment:
right now this is a little redundant. it will show the error in two places. 
![Screenshot 2025-03-06 at 5 54 14 PM](https://github.com/user-attachments/assets/4679ca6f-9dc4-4594-945b-fa0faa5a7a99)